includes/abstracts/abstract.llms.admin.table.php

```diff
@@ -538,66 +535,92 @@ protected function get_table_classes() {
+	}
+
+	/**
+	 * Output HTML for the filters displayed in the head of the table.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	public function output_table_filters_html() {
 		?>
 		<div class="llms-table-filters">
 			<?php foreach ( $this->get_columns() as $id => $data ) : ?>
 				<?php if ( is_array( $data ) && isset( $data['filterable'] ) && is_array( $data['filterable'] ) ) : ?>
 					<div class="llms-table-filter-wrap">
-						<select class="llms-select2 llms-table-filter" id="<?php printf( '%1$s-%2$s-filter', $this->id, $id ); ?>" name="<?php echo $id; ?>">
-							<option value="<?php echo $this->get_filter(); ?>"><?php echo $this->get_filter_placeholder( $id, $data ); ?></option>
+						<select class="llms-select2 llms-table-filter" id="<?php echo esc_attr( sprintf( '%1$s-%2$s-filter', $this->id, $id ) ); ?>" name="<?php echo esc_attr( $id ); ?>">
+							<option value="<?php echo esc_attr( $this->get_filter() ); ?>"><?php echo esc_html( $this->get_filter_placeholder( $id, $data ) ); ?></option>
 							<?php foreach ( $data['filterable'] as $val => $name ) : ?>
-								<option value="<?php echo $val; ?>"><?php echo $name; ?></option>
+								<option value="<?php echo esc_attr( $val ); ?>"><?php echo esc_html( $name ); ?></option>
 							<?php endforeach; ?>
 						</select>
 					</div>
 				<?php endif; ?>
 			<?php endforeach; ?>
 		</div>
 		<?php
-		return ob_get_clean();
 	}
 
+
 	/**
-	 * Get the HTML for the entire table.
+	 * Return the HTML for the entire table.
 	 *
 	 * @since 3.2.0
 	 * @since 3.17.8 Unknown.
 	 * @since 3.37.7 Use correct argument order for implode to fix php 7.4 deprecation.
 	 *
 	 * @return string
+	 * @deprecated [version] Use output_table_html() instead.
 	 */
 	public function get_table_html() {
+		ob_start();
+		$this->output_table_html();
+
+		return ob_get_clean();
+	}
+
+	/**
+	 * Output the HTML for the entire table.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function output_table_html() {
 
 		$classes = $this->get_table_classes();
 
-		ob_start();
 		?>
 		<div class="llms-table-wrap">
 			<header class="llms-table-header">
-				<?php echo $this->get_table_title_html(); ?>
+				<?php $this->output_table_title_html(); ?>
 				<?php if ( $this->is_searchable ) : ?>
-					<?php echo $this->get_table_search_form_html(); ?>
+					<?php $this->output_table_search_form_html(); ?>
 				<?php endif; ?>
 				<?php if ( $this->is_filterable ) : ?>
-					<?php echo $this->get_table_filters_html(); ?>
+					<?php $this->output_table_filters_html(); ?>
 				<?php endif; ?>
 			</header>
 			<table
-				class="<?php echo implode( ' ', $classes ); ?>"
-				data-args='<?php echo wp_json_encode( $this->get_args() ); ?>'
-				data-handler="<?php echo $this->get_handler(); ?>"
-				id="llms-gb-table-<?php echo $this->id; ?>"
+				class="<?php echo esc_attr( implode( ' ', $classes ) ); ?>"
+				data-args='<?php echo esc_attr( wp_json_encode( $this->get_args() ) ); ?>'
```

Comment:
We should be good to esc_attr this even though wp_json_encode() should be safe

Reply:
Agreed. esc_attr and esc_html don't seem to interfere with the json as used here and similar places.